index.js

```diff
@@ -1,11 +1,277 @@
+            value: 'repeat'
+          }
+        ],
+        def: 'single'
+      },
+      endDate: {
+        label: 'End date',
+        type: 'date',
+        if: {
+          dateType: 'consecutive'
+        }
+      },
+      repeatInterval: {
+        label: 'How often does the event repeat?',
+        type: 'select',
+        choices: [
+          {
+            label: 'Every week',
+            value: 'weeks'
+          },
+          {
+            label: 'Every month',
+            value: 'months'
+          }
+        ],
+        if: {
+          dateType: 'repeat'
+        }
+      },
+      repeatCount: {
+        label: 'How many times does it repeat?',
+        type: 'integer',
+        def: 1,
+        if: {
+          dateType: 'repeat'
+        }
+      },
+      description: {
+        type: 'string',
+        label: 'Description',
+        textarea: true,
+        required: true
+      },
+      image: {
+        label: 'Headline photo',
+        type: 'area',
+        options: {
+          max: 1,
+          widgets: {
+            '@apostrophecms/image': {}
+          }
+        },
+        required: false
+      }
+    },
+    group: {
+      basics: {
+        label: 'Basics',
+        fields: [
+          'title',
+          'slug',
+          'description',
+          'image',
+          'startDate',
+          'allDay',
+          'startTime',
+          'endTime'
+        ]
+      },
+      advanced: {
+        label: 'Advanced',
+        fields: ['dateType', 'endDate', 'repeatInterval', 'repeatCount']
+      },
+      meta: {
+        label: 'Meta',
+        fields: ['tags', 'published']
+      }
+    }
+  },
+  handlers(self, options) {
+    return {
+      beforeSave: {
+        async denormalizeDateTimes(req, piece, options) {
+          self.denormalizeDatesAndTimes(piece);
+        }
+      },
+      beforeInsert: {
+        setGroupId(req, piece, options) {
+          // Set groupId on parent if this is a repeating item
+          if (
+            piece.dateType === 'repeat' &&
+            !piece.groupId &&
+            !self._workflowPropagating
+          ) {
+            piece.groupId = self.apos.util.generateId();
+          }
+        }
+      },
+      afterInsert: {
+        async createRepeatItems(req, piece, options) {
+          if (self._workflowPropagating) {
+            // Workflow is replicating this but also its existing
+            // scheduled repetitions, don't re-replicate them and cause problems
+            return;
+          }
+          if (piece.dateType === 'repeat' && piece.aposMode === 'draft') {
+            await self.repeatEvent(req, piece, options);
+          }
+        }
+      },
+      afterPublish: {
+        async publishChildren(req, piece, options) {
+          // If this is a repeating item, publish its children also
+          if (piece.published.dateType === 'repeat' && piece.firstTime) {
+            const existing = await self.findChildren(req, {
```

Comment:
I don't think the `findChildren` method is adding much at this point. Either it should take `req, piece` so we don't have to think about `groupId` when making the query, or we should just move the `find` call and chain `toArray` after it right here.

Reply:
OK - I removed the findChildren method, seemed to be the cleaner solution